### PR TITLE
[Documentation] Update API documentation 

### PR DESCRIPTION
**Pull request by tuka217**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | no|
| BC breaks?      | no|
| Related tickets | based on  #7604 |
| License         | MIT |

Should be merged after #7608, #7602, #7601

### CODE REVIEW

File: docs/api/product_variants.rst [hunk 1]
```diff
@@ -643,6 +646,10 @@ To fully update the product variant with ``code = double-theme-mug`` for the pro
             }
         '
 
+.. warning::
+
+    Do not forget to pass version of the variant. Without this  you will receive a ``409 Conflict`` error.
```
Comment (TheMadeleine) [inline docs/api/product_variants.rst]: `Without this  you` <- double space.

File: docs/api/promotion_coupons.rst [hunk 2]
```diff
@@ -35,10 +21,28 @@ If you request for more detailed data, you will receive an object with the follo
...
++-----------------------+--------------------------------------------------+
+| expiresAt             | The date when the coupon will be no longer valid |
++-----------------------+--------------------------------------------------+
+| usageLimit            | Number of times this coupon has been used        |
++-----------------------+--------------------------------------------------+
+| createdAt             | Date of creation                                 |
++-----------------------+--------------------------------------------------+
+| updatedAt             | Date of last update                              |
++-----------------------+--------------------------------------------------+
+| perCustomerUsageLimit | Date of last update                              |
```
Comment (TheMadeleine) [inline docs/api/promotion_coupons.rst]: `Date of last update`?  ;> U sure? :tada:

File: docs/api/provinces.rst [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+.. code-block:: text
+
+    GET /api/v1/codes/{countryCode}/provinces/{code}
+
++---------------+----------------+---------------------------------------------+
+| Parameter     | Parameter type | Description                                 |
++===============+================+=============================================+
+| Authorization | header         | Token received during authentication        |
++---------------+----------------+---------------------------------------------+
+| countryCode   | url attribute  | Code of country to which province belonging |
```
Comment (TheMadeleine) [inline docs/api/provinces.rst]: `Code of the country to which the province belongs`

File: docs/api/provinces.rst [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
++---------------+----------------+---------------------------------------------+
+| code          | url attribute  | Code of requested province                  |
```
Comment (TheMadeleine) [inline docs/api/provinces.rst]: `Code of the requested province`

File: docs/api/provinces.rst [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
++---------------+----------------+---------------------------------------------+
+
+Example
+^^^^^^^
+
+To see the details of the province with ``code = PL-MZ`` which belongs to country with ``code = PL`` use the below method:
```
Comment (TheMadeleine) [inline docs/api/provinces.rst]: to **the** country

File: docs/api/provinces.rst [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+.. code-block:: bash
+
+     $ curl http://demo.sylius.org/api/v1/countries/PL/provinces/PL-MZ \
+        -H "Authorization: Bearer SampleToken" \
+        -H "Accept: application/json"
+
+.. note::
+
+    The *PL* ans *PL-MZ* codes is just an example. Your value can be different.
```
Comment (TheMadeleine) [inline docs/api/provinces.rst]: `... codes are just examples.`

File: docs/api/provinces.rst [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    DELETE /api/v1/countries/{countryCode}/provinces/{code}
+
++---------------+----------------+---------------------------------------------+
+| Parameter     | Parameter type | Description                                 |
++===============+================+=============================================+
+| Authorization | header         | Token received during authentication        |
++---------------+----------------+---------------------------------------------+
+| countryCode   | url attribute  | Code of country to which province belonging |
++---------------+----------------+---------------------------------------------+
+| code          | url attribute  | Code of requested province                  |
```
Comment (TheMadeleine) [inline docs/api/provinces.rst]: `Code of the requested province`

File: docs/api/provinces.rst [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+.. code-block:: text
+
+    DELETE /api/v1/countries/{countryCode}/provinces/{code}
+
++---------------+----------------+---------------------------------------------+
+| Parameter     | Parameter type | Description                                 |
++===============+================+=============================================+
+| Authorization | header         | Token received during authentication        |
++---------------+----------------+---------------------------------------------+
+| countryCode   | url attribute  | Code of country to which province belonging |
```
Comment (TheMadeleine) [inline docs/api/provinces.rst]: `Code of the country to which the province belongs`

File: docs/api/promotion_coupons.rst (hunk 2 above)
Comment (tuka217) [inline docs/api/promotion_coupons.rst]: 🦄

**Comment by pjedrzejewski** [comment]
Thank you Ania! :)

**Pull request merged**